components/builder-api-client/src/lib.rs

```diff
@@ -380,6 +381,53 @@ impl Client {
     }
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum OriginMemberRole {
+    Member,
+    Maintainer,
+    Administrator,
+    Owner,
+}
+
+impl Default for OriginMemberRole {
+    fn default() -> OriginMemberRole { OriginMemberRole::Member }
+}
+
+impl fmt::Display for OriginMemberRole {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let value = match *self {
+            OriginMemberRole::Member => "member",
+            OriginMemberRole::Maintainer => "maintainer",
+            OriginMemberRole::Administrator => "administrator",
+            OriginMemberRole::Owner => "owner",
+        };
+        write!(f, "{}", value)
+    }
+}
+
+impl FromStr for OriginMemberRole {
+    type Err = Error;
+
+    fn from_str(value: &str) -> result::Result<Self, Self::Err> {
+        match value.to_lowercase().as_ref() {
+            "member" => Ok(OriginMemberRole::Member),
+            "maintainer" => Ok(OriginMemberRole::Maintainer),
+            "administrator" => Ok(OriginMemberRole::Administrator),
+            "owner" => Ok(OriginMemberRole::Owner),
+            _ => Err(Error::BadOriginMemberRole(value.to_string())),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct OriginMemberRoleResponse {
```

Comment:
Do we need this type? Could we implement `PortableText` directly on `OriginMemberRole`?

Reply:
Yep, we need this one. The response that comes from the API endpoint is not a `OriginMemberRole` but a struct that effectively wraps it, hence `OriginMemberRoleResponse`.